### ISSUE TITLE
Demos need a common template

### ISSUE THREAD

**Issue by tomgreever** [bitovi/documentjs#200]
Demo files need to share a common template so they can be styled appropriately. Currently, its not possible to define a common HTML template that all demos share. As a result, I copy/paste all my demos.

One of the templates could be `demos.mustache` which might contain something like: 
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <title>{{demoTitle}}</title>
    <link rel="stylesheet" href="{{demoCSS}}">
</head>

<body>
 <div class="demo-wrapper">
  {{demoContent}}
 </div>
</body>
</html>
```

**Comment by justinbmeyer** [comment]
demo's use an iframe.  We could write to the iframe using the template after XHRing the demo content.

In general, many CanJS demo pages don't include the doctype, html, head, body, etc.  They just look like:

```
<script src="../steal.js"></script>
<script can-autoload>
  stuff
</script>
```

**Comment by rcherny** [comment]
Right, so ... Demos are in an Iframe, is every demo file expected to be a whole HTML file with CSS linked and everything? Now that I type that, it sounds like a dumb question, but frequently I was thinking the demos would be fragments used some other way as part of the application or Web site I'm documenting.

So currently, I have to make each a full file? I'm coming at this from the documentcss side, though I understand the features of documentjs (and hope to use it in the future) as well.

Having some trouble even locating the demo CSS files / templates and such that go into the documentcss @ demos tags ...

**Comment by tomgreever** [comment]
@rcherny they do not have to be in a full HTML file, no. I often use a full file. The need for a common template, though, I think it useful if I want all demos to have a common wrapping container or some other dependency.

> Having some trouble even locating the demo CSS files / templates and such that go into the documentcss @ demos tags ...

I'm not sure what you mean by this? There aren't templates for the demos (which is what this feature request is about). Currently, you create all your demo files yourself.
